thingsearch/service/src/main/java/org/eclipse/ditto/thingsearch/service/common/config/DefaultCustomSearchMetricConfig.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.ditto.thingsearch.service.common.config;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.internal.utils.config.ConfigWithFallback;
+import org.eclipse.ditto.thingsearch.service.placeholders.GroupByPlaceholderResolver;
+import org.eclipse.ditto.thingsearch.service.placeholders.InlinePlaceholderResolver;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+@Immutable
+public final class DefaultCustomSearchMetricConfig implements CustomSearchMetricConfig {
+
+    private final String metricName;
+    private final boolean enabled;
+    private final Duration scrapeInterval;
+    private final List<String> namespaces;
+    private final Map<String, String> groupBy;
+    private final Map<String, String> tags;
+    private final List<FilterConfig> filterConfigs;
+
+    private DefaultCustomSearchMetricConfig(final String key, final ConfigWithFallback configWithFallback) {
+        this.metricName = key;
+        enabled = configWithFallback.getBoolean(CustomSearchMetricConfigValue.ENABLED.getConfigPath());
+        scrapeInterval = configWithFallback.getDuration(CustomSearchMetricConfigValue.SCRAPE_INTERVAL.getConfigPath());
+        namespaces = Collections.unmodifiableList(new ArrayList<>(
+                configWithFallback.getStringList(CustomSearchMetricConfigValue.NAMESPACES.getConfigPath())));
+        groupBy = Collections.unmodifiableMap(new HashMap<>(
```

Comment:
shouldn't this be a `new LinkedHashMap` instead? I assume the "grouping by" is sensitive to the order?